shell/common/shell.cc

```diff
@@ -2023,6 +2033,66 @@ bool Shell::OnServiceProtocolReloadAssetFonts(
   return true;
 }
 
+void Shell::AddView(int64_t view_id, const ViewportMetrics& viewport_metrics) {
+  TRACE_EVENT0("flutter", "Shell::AddView");
+  FML_DCHECK(is_set_up_);
+  FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
+  if (view_id == kFlutterImplicitViewId) {
+    FML_DLOG(ERROR) << "Unexpected request to add the implicit view #"
+                    << kFlutterImplicitViewId
+                    << ". This view should never be added. This call is no-op.";
```

Comment:
Should this be an assertion that exits? FYI, `FML_DCHECK` can log a message. For example, see:

https://github.com/flutter/engine/blob/9c4db1788f915b85d9f92fa5860a58b40884ae65/fml/shared_thread_merger.cc#L79

Reply:
I was planning to make AddView and RemoveView more tolerant to implicit views by returning with a log. But making them assert works too.